projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -3270,8 +3406,15 @@ public static function validate_custom_service_id( $value, $request, $param ) {
 		$sharer   = new Sharing_Service();
 		$services = array_keys( $sharer->get_all_services() );
 
-		if ( ! empty( $value ) && ! in_array( $value, $services ) ) {
-			return new WP_Error( 'invalid_param', sprintf( esc_html__( '%s is not a registered custom sharing service.', 'jetpack' ), $param ) );
+		if ( ! empty( $value ) && ! in_array( $value, $services, true ) ) {
```

Comment:
We could simplify this code a bit, no need to do effectively `in_array( $value, array_keys( ... ) )` when we're not using the list of keys for anything else either.
```suggestion
		$services = $sharer->get_all_services();

		if ( ! empty( $value ) && ! isset( $services[ $value ] ) ) {
```

Reply:
👍 65273c828780fa47bdae4592ab80d85b36729792